### PR TITLE
implement vectorized evaluation for:builtinJSONArraySig

### PR DESCRIPTION
**Pull request by icditwang**
### What problem does this PR solve? 
implement vectorized evaluation for:builtinJSONArraySig

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArraySig-VecBuiltinFunc-8            	    5000	    333547 ns/op	  172065 B/op	    6145 allocs/op

BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArraySig-NonVecBuiltinFunc-8         	    5000	    343952 ns/op	  147488 B/op	    6144 allocs/op
```
### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_json_vec_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ var vecBuiltinJSONCases = map[string][]vecExprBenchCase{
 	ast.JSONExtract:      {},
 	ast.JSONLength:       {},
 	ast.JSONType:         {{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETJson}}},
-	ast.JSONArray:        {},
+	ast.JSONArray:        {{retEvalType: types.ETJson, childrenTypes: []types.EvalType{types.ETJson}}},
```
Comment (wshwsh12) [inline expression/builtin_json_vec_test.go]: Please add more test cases about many args. Only one arg is not enough.

Comment (icditwang) [reply expression/builtin_json_vec_test.go]: like this?
like this?:`{retEvalType: types.ETJson, childrenTypes: []types.EvalType{
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
			types.ETJson，types.ETJson
		}}`

Comment (wshwsh12) [reply expression/builtin_json_vec_test.go]: I think that add a test about three arguments is also needed.  Rest LGTM.

Comment (icditwang) [reply expression/builtin_json_vec_test.go]: done!

**Pull request merged**